bench/BenchIndices.hs

```diff
@@ -0,0 +1,136 @@
+{-# LANGUAGE BangPatterns        #-}
+-- |
+-- Copyright   : (c) 2011 Simon Meier
+-- License     : BSD3-style (see LICENSE)
+--
+-- Maintainer  : Simon Meier <iridcode@gmail.com>
+-- Stability   : experimental
+-- Portability : tested on GHC only
+--
+-- Benchmark all 'Builder' functions.
+module Main (main) where
+
+import           Data.Foldable                         (foldMap)
+import           Data.Monoid
+import           Data.String
+import           Gauge
+import           Prelude                               hiding (words)
+import           GHC.Word                              (Word8, eqWord8)
+
+import qualified Data.ByteString                       as S
+import qualified Data.ByteString.Unsafe                as S
+
+
+------------------------------------------------------------------------------
+-- Benchmark
+------------------------------------------------------------------------------
+
+
+
+-- lines of 200 letters from a to e, followed by repeated letter f
+absurdlong :: S.ByteString
+absurdlong = S.replicate 200 0x61 <> S.singleton 0xa
+          <> S.replicate 200 0x62 <> S.singleton 0xa
+          <> S.replicate 200 0x63 <> S.singleton 0xa
+          <> S.replicate 200 0x64 <> S.singleton 0xa
+          <> S.replicate 200 0x65 <> S.singleton 0xa
+          <> S.replicate 999999 0x66
+{-# NOINLINE absurdlong #-}
```

Comment:
Why is it important not to inline?

Reply:
I had attempted to copy the pattern used in `BenchAll.hs` whereby constant values used as arguments to benchmarked functions were given NOINLINE pragmas to facilitate memoization. In hindsight this wasn't motivated by any specific intention on my part, so for the time being I removed the pragma. There might be a better way of ensuring that the benchmark time doesn't include any overhead for recomputation of `absurdlong`, if indeed such an overhead is even present in the current version of this benchmark module.

Reply:
I believe it's a responsibility of `nf` to evaluate its second argument before benchmarking.